zip-0227.rst

```diff
@@ -1,8 +1,359 @@
+bridging purposes.
+
+Use Cases
+=========
+
+The design presented in this ZIP enables issuance of shielded assets in various modes:
+
+- The issuer knows in advance the receivers of the issued asset.
+- The asset is of non-fungible type, where each asset type can be made part of a single “series”
+- The supply of the asset is limited or not
+- The assets can be wrapped versions of assets in other chains (as long as there is a bridge that supports it)
+
+See the Concrete Applications section for more details.
+
+Requirements
+============
+
+- Any user of the Zcash block chain can issue custom assets on chain
+- The issuance mechanism should enable tracking publicly the supply of the assets on the Zcash block chain
+- A user without knowledge of the private key of the issuer of a specific asset should not be able to issue that asset or change the attributes of such asset
+- The asset identification should be unique and different issuer keys should not be able to generate the same asset identifier
+
+
+Specification
+=============
+
+As explained, the issuance protocol allows for a single issuance action to be sent to many receivers by mean of generating as many output notes as distinct recipients. Furthermore, every bundle can contain many issuance actions. Thus we divide the protocol spec in two sections.
+
+Overview 
+--------
+
+In short, this specification enables only *transparent* issuance, since as a first step, transparency is important to properly test the applications that will be most used in the Zcash ecosystem, while enabling tracking of supply of assets. Any user of the blockchain can issue new asset types, yet only a single issuer (a set of keys) can issue that specific asset type, as the asset identifiers are tied to the issuer keys.
+
+The protocol was designed with three main properties in mind:
+- at the time of issuance, the issuer can already allocate all the tokens to the corresponding owners by creating the corresponding (shielded) output notes to the respective addresses. As is implied, the issuance mechanism is itself transparent, but the allocation is totally private
+- in a single issuance bundle, the issuer can publish many issuance actions, or in other words, create multiple custom assets
+- every issuance action contains the `finiteIssuance` boolean that defines whether that specific custom asset can have further tokens issued or not
+
+The issuance mechanism is broad enough for issuers to either create native assets on Zcash (i.e.: assets that originate on the Zcash blockchain), as well as for institutions to create bridges from other chains and import wrapped assets. In fact, the protocol described below enables, what we hope will be a useful set of, applications. The design decision is not to have a chosen name to describe the custom asset, but to delegate it to an off-chain mapping, as this would imply a land-grab “war”.
```

Comment:
```suggestion
The issuance mechanism is broad enough for issuers to either create native assets on Zcash (i.e.: assets that originate on the Zcash blockchain), as well as for institutions to create bridges from other chains and import wrapped assets. In fact, the protocol described below enables what we hope will be a useful set of applications. The design decision is not to have a chosen name to describe the custom asset, but to delegate it to an off-chain mapping, as this would imply a land-grab “war”.
```

Reply:
Done.